catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/util/ProjectSetupUtils.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.util;
+
+import org.catrobat.catroid.content.BroadcastScript;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.bricks.SetVariableBrick;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+
+public final class ProjectSetupUtils {
+	// Suppress default constructor for noninstantiability
+	private ProjectSetupUtils() {
+		throw new AssertionError();
+	}
+
+	public static Script addBroadcastScriptSettingVariable(Sprite sprite, String message, UserVariable userVariable,
+			double value) {
+		Script broadcastScript = new BroadcastScript(message);
+		SetVariableBrick setVariableBrickAfterBroadcast = new SetVariableBrick(new Formula(value), userVariable);
+		broadcastScript.addBrick(setVariableBrickAfterBroadcast);
+		sprite.addScript(broadcastScript);
+		return broadcastScript;
+	}
+
+	public static Script addSpriteWithScript(Project project, String spriteName, Script script) {
+		Sprite sprite = new Sprite(spriteName);
+		project.getDefaultScene().addSprite(sprite);
+		return addScriptToSprite(script, sprite);
+	}
+
+	public static Script addScriptToSprite(Script script, Sprite sprite) {
```

Comment:
why tho?
i dont see the point of this function, caller has to know script and sprite and can just call sprite.addScript himself.